dbms/src/Formats/FormatFactory.cpp

```diff
@@ -96,6 +119,13 @@ void FormatFactory::registerOutputFormat(const String & name, OutputCreator outp
     target = output_creator;
 }
 
+void FormatFactory::registerChunkGetter(const String & name, ChunkCreator chunk_creator)
```

Comment:
So is it a getter or a creator? I'd name it something like FileSegmentationEngine.

Reply:
Renamed